src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -102,25 +102,29 @@ def convert_metadata_to_sp_type(
                 "Vectors are not supported by your connector: Please update it to a newer version"
             )
 
-        if metadata.fields is None:
+        fields = getattr(metadata, "fields", None)
```

Comment:
is this change required by mypy, or ResultMetadata import?

Reply:
can we do this refactoring separately?